doc-changelog/action.yml

```diff
@@ -91,7 +99,60 @@ runs:
       run: |
         python -m pip install --upgrade pip towncrier==${{ inputs.towncrier-version }}
 
+    - name: "Check pull-request title follows conventional commits style"
+      id: lowercase
+      if: ${{ inputs.use-labels == 'false' && ((github.event_name == 'pull_request_target') || (github.event_name == 'pull_request')) }}
+      uses: amannn/action-semantic-pull-request@v5
+      env:
+        GITHUB_TOKEN: ${{ inputs.token }}
+      continue-on-error: true
+
+    - name: "Check pull-request title follows conventional commits style with upper case"
+      id: caps
+      if: ${{ inputs.use-labels == 'false' && ((github.event_name == 'pull_request_target') || (github.event_name == 'pull_request')) }}
+      uses: amannn/action-semantic-pull-request@v5
+      env:
+        GITHUB_TOKEN: ${{ inputs.token }}
+      with:
+        types: |
+          BUILD
+          CHORE
+          CI
+          DOCS
+          FEAT
+          FIX
+          PERF
+          REFACTOR
+          REVERT
+          STYLE
+          TEST
+      continue-on-error: true
+
+    - name: "Check failures"
+      if: steps.lowercase.outcome == 'failure' && steps.caps.outcome == 'failure'
+      shell: bash
+      run: exit 1
+
+    - name: "Get conventional commit type from title"
+      if: ${{ inputs.use-labels == 'false' }}
+      env:
+        PR_TITLE: ${{ github.event.pull_request.title }}
+      shell: python
+      run: |
+        import os
+
+        pr_title = f"""{os.environ.get('PR_TITLE')}"""
+        colon_index = pr_title.index(":")
+        cc_type = '"'+pr_title[:colon_index]+'"'
+
+        # Get the GITHUB_ENV variable
+        github_env = os.getenv('GITHUB_ENV')
+
+        with open(github_env, "a") as f:
+            f.write(f'CC_TYPE={cc_type}')
```

Comment:
Can you rewrite this into a python file and run it like in `doc-build/parse_doc_conf.py`.
If possible, do the same for other python parts of this action !
This would be another step to #459 and we might want to leverage this in other actions.